app/models/place.rb

```diff
@@ -27,5 +30,15 @@ def update_address(place, raw_place)
   def set_slug
     self.slug = name.parameterize
   end
-   
+
+  def self.slug_for_show(place_id)
+    @client = GooglePlaces::Client.new(ENV['GOOGLE_KEY'])
+    raw_place = @client.spot(place_id)
+    Place.from_google_api(raw_place).slug
+  end
+
+  def self.query_for_id(search_text)
+    @client = GooglePlaces::Client.new(ENV['GOOGLE_KEY'])
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.